plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -244,8 +310,17 @@ public static IcebergColumnHandle getColumnHandle(NestedField column, TypeManage
 
     public static Schema schemaFromHandles(List<IcebergColumnHandle> columns)
     {
+        int maxParentId = columns.stream().map(column -> column.getId()).max(Comparator.naturalOrder()).get();
+
+        AtomicInteger nextNestedFieldId = new AtomicInteger(maxParentId + 1);
         List<NestedField> icebergColumns = columns.stream()
-                .map(column -> NestedField.optional(column.getId(), column.getName(), toIcebergType(column.getType())))
+                .map(column -> {
+                    org.apache.iceberg.types.Type type = toIcebergType(column.getType());
+                    if (!type.isPrimitiveType()) {
+                        type = TypeUtil.assignFreshIds(type, nextNestedFieldId::getAndIncrement);
```

Comment:
This doesn't seem right to me.
Doing tricks like this can get us into trouble (there are some syntethic columns - see `org.apache.iceberg.MetadataColumns#FILE_PATH` ) which have the id set to `Integer.MAX_VALUE - 1`. Incrementing this value to get artificially a new field id for the row can lead to problems.

Reply:
I was looking for where `schemaFromHandles` is being used and found the following:

- deletes handling in `IcebergPageSourceProvider`
- bucketing function in `IcebergNodePartitioningProvider` 

In both places we do have `schema` available.

Please investigate whether using `schema` in `schemaFromHandles` would be possible.
If yes, we could work with `TypeUtil.indexParents(schema)` to get only the extra "row" contents we need from the schema.

Reply:
This is related to some changes in https://github.com/trinodb/trino/pull/14837

Maybe we can pull some of those changes in here

Reply:
Thanks, @findinpath | @alexjo2144 for pointing this out. I have added some changes from https://github.com/trinodb/trino/pull/14837 and now I do not need to reassign indexes.

IMO, It will be better if https://github.com/trinodb/trino/pull/14837 gets merged first because the current PR contains changes, that are unrelated to the supporting partitioning field.